tests/unit/query_graph/test_adapter.py

```diff
@@ -106,3 +106,24 @@ def test_will_pivoted_column_name_be_quoted(column_name, will_be_quoted):
     Test will_pivoted_column_name_be_quoted for SnowflakeAdapter
     """
     assert SnowflakeAdapter.will_pivoted_column_name_be_quoted(column_name) is will_be_quoted
+
+
+@pytest.mark.parametrize(
+    "percent, expected_format",
+    [
+        (10.01, "10.01"),
+        (10.0, "10"),
+        (10, "10"),
+        (0.1, "0.1"),
+        (0.000001, "0.000001"),
+        (0.000000001, "0.000000001"),
+    ],
+)
+def test_tablesample_percentage_formatting(percent, expected_format):
+    """
+    Test the percentage in TABLESAMPLE is not formatted using scientific notation because that is
+    not supported in some engines like Spark
+    """
+    out = SnowflakeAdapter.tablesample(select("*").from_("A"), percent).sql()
```

Comment:
nit: can consider making a `class BaseTestAdapater(BaseAdapater)` for testing the base class methods so that it's clearer that these functions are coming from the base class.